front_end/src/components/charts/numeric_chart.tsx

```diff
@@ -174,14 +174,15 @@ const NumericChart: FC<Props> = ({
       cursorLabelComponent={<ChartCursorLabel positionY={height - 10} />}
       onCursorChange={(value: CursorCoordinatesPropType) => {
         if (typeof value === "number" && onCursorChange) {
-          const closestForecast = aggregation.history.reduce((prev, curr) => {
-            if (curr.start_time <= value) {
-              return curr.start_time > prev.start_time ? curr : prev;
-            }
-            return prev;
-          });
-
-          onCursorChange(closestForecast.start_time);
+          const timestamps = uniq([
+            ...aggregation.history.map((f) => f.start_time),
+            ...aggregation.history.map((f) => f.end_time ?? f.start_time),
+            actualCloseTime ?? Date.now() / 1000,
+          ]).sort((a, b) => a - b);
```

Comment:
Do we actually have to run it on every cursor change event? It may cause performance issues. I think we could move it into a separate variable and additionally, utilize `useMemo` hook to reduce variable recalculations:

```
  const timestamps = useMemo(
    () =>
      uniq([
        ...aggregation.history.map((f) => f.start_time),
        ...aggregation.history.map((f) => f.end_time ?? f.start_time),
        actualCloseTime ?? Date.now() / 1000,
      ]).sort((a, b) => a - b),
    [actualCloseTime, aggregation.history]
  );
  
  ...
  
        onCursorChange={(value: CursorCoordinatesPropType) => {
        if (typeof value === "number" && onCursorChange) {
          onCursorChange(
            timestamps[timestamps.findIndex((t) => t > value) - 1]
          );
        }
      }}
  
  ```

Reply:
Great point!